index.js

```diff
@@ -215,6 +527,63 @@ class Game {
     this.moves.push(move);
   }
 
+  moveIntoCheck(arr, movingPience) {
+    for (let y = 0; y < 8; y++) {
+      for (let x = 0; x < 8; x++) {
+        const opponentPiece = this.board.getSquareContent(y, x);
+        if (
+          opponentPiece !== null &&
+          opponentPiece.color !== movingPience.color
+        ) {
+          for (let i = 0; i < opponentPiece.moves.length; i++) {
+            const piece = this.board.getSquareContent(
+              opponentPiece.moves[i].row,
+              opponentPiece.moves[i].file
+            );
+            if (piece !== null && piece.name !== "King") {
+              arr.push(false);
+            } else if (piece !== null && piece.name === "King") {
+              arr.push(true);
+              piece.isCheck();
+            }
+          }
+        }
+      }
+    }
+    return arr;
+  }
+
+  didCheck(piece) {
+    for (let i = 0; i < piece.moves.length; i++) {
+      const pieceInView = this.board.getSquareContent(
+        piece.moves[i].row,
+        piece.moves[i].file
+      );
+      if (
+        pieceInView !== null &&
+        piece.color !== pieceInView.color &&
+        pieceInView.name === "King"
+      ) {
+        pieceInView.isCheck();
+      }
+    }
```

Comment:
this is a pretty easy pitfall to run into. every time we reference piece.moves it'll re-run the moves function. Let's save some computing power by avoiding the recomputaiton.
```suggestion
    for (let move of piece.moves) {
      const pieceInView = this.board.getSquareContent(
        move.row,
        move.file
      );
      if (
        pieceInView !== null &&
        piece.color !== pieceInView.color &&
        pieceInView.name === "King"
      ) {
        pieceInView.isCheck();
      }
    }
```